File: 1448-count-good-nodes-in-binary-tree/1448-count-good-nodes-in-binary-tree.js
/**
 * Definition for a binary tree node.
 * function TreeNode(val, left, right) {
 *     this.val = (val===undefined ? 0 : val)
 *     this.left = (left===undefined ? null : left)
 *     this.right = (right===undefined ? null : right)
 * }
 */
/**
 * @param {TreeNode} root
 * @return {number}
 */
var goodNodes = function (root) {
  let count = 0;

  function dfs(root, maxValue) {
    if (!root) return;
    
    if (root.val >= maxValue) count++;
    
    maxValue = Math.max(maxValue, root.val);
    
    dfs(root.left, maxValue);
    dfs(root.right, maxValue);
  }
  
  dfs(root, root.val);
  
  return count;
}

File: 1448-count-good-nodes-in-binary-tree/1448-count-good-nodes-in-binary-tree.py
# Definition for a binary tree node.
# class TreeNode:
#     def __init__(self, val=0, left=None, right=None):
#         self.val = val
#         self.left = left
#         self.right = right
class Solution:
  def goodNodes(self, root: TreeNode) -> int:
    self.count = 0

    def dfs(root, max_val):
      if not root:
        return

      if root.val >= max_val:
        self.count += 1

      max_val = max(max_val, root.val)

      dfs(root.left, max_val)
      dfs(root.right, max_val)

    dfs(root, root.val)

    return self.count
